### PR TITLE
Add panes for layers

### LINKED ISSUE #52: Refactor Leaflet Layer Organization
## Overview

Currently, all layers are directly added to the map:

image

which makes them all indistinguishable nodes in the DOM, making layer-specific CSS styling impossible.

Use [Layer Groups](https://leafletjs.com/reference.html#layergroup) for semantic organization of map shapes. This will make for cleaner DOM, and allow CSS targeting if needed.

Once this is done, use the following order of layers, with the first being foremost:

- User Polygon
- Reference Image Layer
- Parcel Data
- Municipal Boundary
- Basemap

When multiple are visible, the topmost one should receive all mouse events.

### PR DESCRIPTION
**Pull request by mstone121**
## Overview

Putting each layer in a pane allows for more control over individual layers because leaflet
creates separate HTML elements for each pane. They can be referenced by the class name `leaflet-{pane label}-pane`. For example, the basemap pane can be selected with `leaflet-basemap-pane`.

Closes #52 

### Demo

Screen Shot 2022-09-02 at 11 41 59 AM

### Notes

I was not able to get the reference layer to use a pane. I tried wrapping it in a `featureGroup` like the user polygon layer, but removed all the edit controls and the image was non-interactive. For now, it seems to work well enough to place all the other layers around that one.

TODO:  From #52: 
> When multiple are visible, the topmost one should receive all mouse events

As is, things seem to be working well. Edit handles can be selected over other layers. The only problem is that you cannot add a polygon over the reference image. This is probably due to this line from Leaflet Distortable Image:
https://github.com/publiclab/Leaflet.DistortableImage/blob/main/dist/leaflet.distortableimage.css#L6

## Testing Instructions

Adding this feature allows the layers to be ordered based on their zLevel. Previously the municipal boundaries layer would show above the reference image. To test this:

- Toggle both the reference image and the municipal boundaries layers on
- Move the reference image to some municipal boundaries and ensure they appear above them.

## Checklist

- [x] `fixup!` commits have been squashed
- [x] `CHANGELOG.md` updated with summary of features or fixes, following [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines
- [x] `README.md` updated if necessary to reflect the changes
- [x] CI passes after rebase

### CODE REVIEW

**Comment by mstone121** [comment]
@rajadain I didn't end up using layer groups as described in #52, so I wanted to check in with you before proceeding with the click event stuff.

**Comment by rajadain** [comment]
Taking a quick look

File: src/app/src/constants.js [hunk 1]
```diff
@@ -25,3 +25,11 @@ export const NC_SOUTH = 33.851169;
 
 export const PARCELS_LAYER_URL =
     'https://services.nconemap.gov/secure/rest/services/NC1Map_Parcels/MapServer';
+
+// https://leafletjs.com/reference.html#map-mappane
+export const PANES = {
+    basemap: { label: 'basemap', zIndex: 200 },
+    parcels: { label: 'parcels', zIndex: 210 },
```
Comment (rajadain) [inline src/app/src/constants.js:32]: Consider making these keys UPPER CASE to match the `DATA_LAYERS`

**Review by rajadain — COMMENTED** [review]
The pane implementation looks good! I don't think we need to pursue layer groups for this.

I'm okay with being able to manipulate the reference layer image below the polygon:

https://user-images.githubusercontent.com/1430060/188660603-b428d963-17b8-4604-9a7f-f75ec3ff8c77.mp4

It is clear enough which layer is being manipulated.

We may need some UX guidance when it comes to the distortion tools, because while the reference image layer is beneath the polygon, the distortion tools are not. But if they do go beneath, should they still be clickable? Tagging @gatesgodin for recommendations about that.

Also, I noticed that the Welcome page is crashing:

image

We likely need to wrap the `DefaultBasemap` there within `MapPanes` as well, in which case it may make more sense to move `MapPanes` from `DrawMap/` to `Layers/`.

**Comment by gatesgodin** [comment]
> 

@rajadain I'm incline to have the reference layer and tools layered above the boundary polygon. They can use the opacity to set the transparency lower. The user would then toggle the reference layer on/off and adjust polygon accordingly.

**Comment by rajadain** [comment]
Hmm, but if the reference layer is in front of the polygon, they can't edit the polygon while looking at the reference layer. Wouldn't editing a transparent polygon on top of a reference layer be easier?

**Comment by rajadain** [comment]
From a conversation with @mstone121 it seems that all interactive parts (distortion toolbar, polygon vertices, reference image vertices) are drawn on a different pane than the drawn parts (the polygon, the reference image), which makes it difficult to put the distortion toolbar behind the polygon layer.

Another thing we could do is see if we can make the distortion tools only visible when the reference layer is in "focus", but that may make the tool more complicated, as it would add the concept of an active or inactive layer.

**Comment by mstone121** [comment]
> Another thing we could do is see if we can make the distortion tools only visible when the reference layer is in "focus", but that may make the tool more complicated, as it would add the concept of an active or inactive layer.

This is actually sort of built into the tool already. If you click away from the Reference Image the edit tools should disappear, but the image is still visible.

**Comment by rajadain** [comment]
Ah nice! @gatesgodin let's see if the three of us can huddle tomorrow morning on this to set the immediate direction.

**Comment by gatesgodin** [comment]
I am okay with the active / inactive "focus" function as long as the reference layer is clickable. The tools would disappear by clicking away and re-appear when clicking on the reference layer. Once the reference image is aligned correctly on the map, the tools will no longer be necessary. I am also imagining that they would toggle the boundary off when adjusting the reference layer. So I think as you have it now is pretty close to that functionality.

Sounds good, let's huddle in the morning! @mstone121 @rajadain

**Comment by rajadain** [comment]
Summary of discussion:

- We'll keep the implementation above as is, with the reference image layer being beneath the polygon, and the distortion toolbar only becoming visible when the reference image layer is selected
- This implementation gets us most of the way there, can make little tweaks later
- There is no current implementation in the works for the "popout" feature of the reference image layer. That will be tested by simulation in the user research sessions, and if deemed worthy, will be wireframed first, and implemented after.

**Comment by rajadain** [comment]
Will take another look once rebased

**Comment by rajadain** [comment]
Taking a look

**Review by rajadain — COMMENTED** [review]
This works great. I can draw polygons on top of the Parcel Data layer now, which is great.

The only thing still off is that the Municipal Boundary labels are on top of the Reference Image Layer:

image

Would recommend seeing if that can be easily fixed.

File: src/app/src/components/Sidebar.js [hunk 2]
```diff
@@ -34,13 +35,15 @@ const marginLeft = 4;
...
 export default function Sidebar() {
     return (
-        <Flex direction='column'>
-            <TitleBar />
-            <Divider />
-            <ReferenceLayers />
-            <Divider />
-            <BasemapLayers />
-        </Flex>
+        <Container ml={marginLeft} maxW='xs' bg='gray.700' p={0}>
```
[the remark this answers is not in the document]
Comment (mstone121) [reply src/app/src/components/Sidebar.js]: https://github.com/azavea/iow-boundary-tool/pull/57/commits/2d71e4acee43b1ce5a63fe5e705a141c2913c265

**Comment by mstone121** [comment]
I added a pane for the labels in https://github.com/azavea/iow-boundary-tool/pull/57/commits/e6ffd438b00d57091da972a769996d7e11b44e46.

**Comment by rajadain** [comment]
Taking another look

**Comment by rajadain** [comment]
I can't seem to draw a polygon over the Municipal Boundary Layer either. Have you seen this? I only see notes about this for the Reference Image Layer

**Comment by mstone121** [comment]
> I can't seem to draw a polygon over the Municipal Boundary Layer either. Have you seen this? I only see notes about this for the Reference Image Layer

Yep. It happened to me too. Fixed [here](01727280540b72f68c653e834fc8406d2f703b51).

**Review by rajadain — APPROVED** [review]
+1 tested, nice work! I like the refactored component arrangement as well. This is working well.

**Pull request merged**